### PR TITLE
Dowload files support

### PR DESCRIPTION
**Pull request by biegehydra**
Additions
 - Two additional configurations
   - allowDownloadAll - Permits users to zip an entire folder (non recursive) and download it
   - allowDownload - Permits users to download individual files
- 1 additional dependency
  - For zipping files I used JSZip

DownloadAll

### CODE REVIEW

**Comment by biegehydra** [comment]
@qoomon Anything need fixed for this to be brought in?

**Comment by qoomon** [comment]
sorry for the late response, I'll try to review it this weekend

File: index.html [hunk 1]
```diff
@@ -596,6 +622,83 @@ <h2 class="subtitle" v-html="config.subtitleHTML"></h2>
...
+
+          try {
+            await Promise.all(promises);
+
+            zip.generateAsync({ type: 'blob' }).then(content => {
+                // content is the zipped file
+                const urlObject = window.URL || window.webkitURL;
+                const objectUrl = urlObject.createObjectURL(content);
+
+                const link = document.createElement('a');
```
Comment (qoomon) [inline index.html:652]: please move the download logic to a separate function

File: index.html [hunk 2]
```diff
@@ -596,6 +622,83 @@ <h2 class="subtitle" v-html="config.subtitleHTML"></h2>
...
+        },
+        async downloadS3Item(fileUrl, fileName) {
+          try {
+            let response = await fetch(fileUrl);
+            let data = await response.blob();
+            let urlObject = window.URL || window.webkitURL;
+            let objectUrl = urlObject.createObjectURL(data);
+
+            let link = document.createElement('a');
+            link.href = objectUrl;
```
Comment (qoomon) [inline index.html:683]: please move the download logic to a separate function

**Comment by qoomon** [comment]
Hi, thanks for the PR.
Some opinion brainstorming:
- I don't think we need the option `allowDownload`
- We need to have a progress bar for file downloading status
- Mabe `Download All`should download really all files including paginated files
- Maybe we should add checkboxes to mark which files we want to download
- We should add a dialogue to ensure someone really wants to download X files with a size of Y
- What about an option to download all files including subdirectories?
- What about a file count or size limit?

WDYT?

**Comment by biegehydra** [comment]
> * I don't think we need the option `allowDownload`
> * We need to have a progress bar for file downloading status
> * Mabe `Download All`should download really all files including paginated files
> * Maybe we should add checkboxes to mark which files we want to download
> * We should add a dialogue to ensure someone really wants to download X files with a size of Y
> * What about an option to download all files including subdirectories?
> * What about a file count or size limit?
> 
> WDYT?

I think they're all good ideas. I'm not that familiar with Vue or Buefy though so adding all these features will be difficult. I'll take a wack at some of it though when I get the chance.

**Comment by Illia-M** [comment]
@qoomon can we update this PR and merge? Really useful for my use cases when people want download folder or all files in folder instead of walking 1000+ files

**Comment by qoomon** [comment]
@Illia-M Latest version (3.1.0) does support download all files from current page.

@biegehydra thanks a lot for this PR

**Pull request merged**